### PR TITLE
ticket-815: use nearest integer for discrete parameters

### PR DESCRIPTION
**Pull request by pkienzle**
**raspberry**: Don't know if it should be restricted to an integer number of rasps; currently it is a floating point surface fraction.

**star polymer**: Don't know if it should be restricted to an integer number of arms; currently it is a floating point value, allowing fractional arms.

**lamellar_stack_paracrystal**:  handles non-integral number of layers; can the other lamellar models be similarly extended?

**multilayer vesicle**: modified equation formatting and changed some of the text, so needs a doc reviewer

**lamellar_stack_paracrystal, linear_pearls, stacked_disks**:  changed the cutoff to one layer/perl/oreo because zero was giving me nan values: 

Models changed:

    core_multi_shell
    lamellar_hg_stack_caille
    lamellar_stack_caille
    lamellar_stack_paracrystal
    linear_pearls
    multilayer_vesicle
    onion
    pearl_necklace
    raspberry
    rpa
    spherical_sld
    stacked_disks
    star_polymer
    unified_power_Rg

### CODE REVIEW

**Comment by ajj** [comment]
In general, models where the math works with non-integer values should not force the value to an integer. The user must decide how to deal with it and the documentation should make clear that it should be integer.

Other lamellar models could probably be modified to handle non-integer numbers of layers with a  weighted sum. Docs would have to clearly indicate that this was being done.

**Comment by pkienzle** [comment]
The changes are mostly to make sure that float to int conversion is consistent across models (I use rounding) and that form volume uses integer if the form function uses integer.  This will result in a number of models producing different results from before for non-integer inputs.  Models which supported non-integer inputs in 3.x will continue to support non-integer inputs.

Here are the models affected by this patch:
```
    linear_pearls: implicit int via for loop
    multilayer_vesicle: implicit int via for loop
    stacked_disks: implicit int via for loop

    core_multi_shell: was forcing int via ciel
    lamellar_hg_stack_caille: was forcing int via trunc
    lamellar_stack_caille: was forcing int via trunc
    pearl_necklace: was forcing int via round

    onion: int via multiplicity
    rpa: int via multiplicity
    spherical_sld: int via multiplicity
    unified_power_Rg: int via multiplicity

    lamellar_stack_paracrystal: correcting for non-integer portion
    raspberry: not forcing int
    star_polymer: not forcing int

    flexible_cylinder: code cleanup; shouldn't be part of this ticket
```

**Comment by pkienzle** [comment]
Updates in response to Jeff's code review: fix up some integer/float conversions.  Also set the bounds for the pearl necklace and lamellar stack caille models to [1, inf] rather than [0, inf] since the code is not correct for n=0.

**Comment by butlerpd** [comment]
So with regard to this particular pull request then, as I understand it, **no** change is being made to the behavior of models regarding whether or not they restrict parameter to integers or not.  The only change is **HOW** that integer is calculated? Also some models restricted parameters to integers but then used not integers to calculate the volume (presumably because volume was calculated before the integer rounding/truncation was done)? This last clearly would have been just wrong.  So to that extent results may differ somewhat from previous versions but mostly "more correct?" 

In other words the larger issues of integer parameters are being left for further discussion but this just cleans things up a bit at the expense of possibly having some different answers from previous versions?

If so, and Jeff has looked at the code I would argue this should be pulled.  Will ask for concurrence from Andrew and/or Richard.
